### PR TITLE
Follow new guidelines for 3rd party repositories on Ubuntu/Debian

### LINKED ISSUE #947: Ubuntu Installation instructions wont work for Linux Mint.
## Description

The installation instructions https://qgis.org/en/site/forusers/alldownloads.html don't include Linux Mint, and anyone exactly following the "Debian/Ubuntu" instructions will get a failure. 

The instructions say 

```
sudo add-apt-repository "deb https://qgis.org/ubuntu $(lsb_release -c -s) main"
```

which on a Mint system puts the Mint codename into the repository line ("uma"). What is needed here is the *ubuntu* codename ("focal").

```
$ echo deb https://qgis.org/ubuntu $(lsb_release -c -s) main
deb https://qgis.org/ubuntu uma main
```

I'm not sure of an absolutely reliable way of getting the Ubuntu codename but possibly sourcing `/etc/os-release` will set `$UBUNTU_CODENAME` on both a Mint "uma" system and a vanilla Ubuntu "focal" system. The alternative is adding a note to the page saying something like "Linux Mint users - find your Ubuntu version codename and put it here". Or maybe `uma` can be linked to `focal` on the server?

Adding the repository with "focal" instead of "uma" (and then doing the usual update/install stuff) results in a working QGIS 3.22 :)

Page URL:  https://qgis.org/en/site/forusers/alldownloads.html

### PR DESCRIPTION
**Pull request by nicogodet**
Here is proposed changes to follow guidelines for third party repositories on Ubuntu and Debian according to https://wiki.debian.org/DebianRepository/UseThirdParty.

Fixes #947 too

### CODE REVIEW

**Review by eshattow — COMMENTED** [review]
sources.list(5) says /usr/share/keyrings for keyrings managed by packages and /etc/apt/keyrings for keyrings managed by the system operator. This was added 2022-05-23 to the https://wiki.debian.org/DebianRepository/UseThirdParty guide "Recommend /etc/apt/keyrings for locally-managed keys, as per apt 2.4.0 changelog".

Also from sources.list(5) says of the Deb822 ".sources" format "It is intended to make this format gradually the default format, deprecating the previously described one-line-style format, as it is easier to create, extend and modify for humans and machines alike..." so would recommend that over continuing with ".list" format

**Comment by nicogodet** [comment]
@jef-n Key does not need to be dearmored anymore ?

**Comment by jef-n** [comment]
> @jef-n Key does not need to be dearmored anymore ?

Not the keyring.

```
$ file qgis-2022.gpg.key qgis-archive-keyring.gpg
qgis-2022.gpg.key:        PGP public key block Public-Key (old)
qgis-archive-keyring.gpg: GPG keybox database version 1, created-at Tue Aug  9 16:14:40 2022, last-maintained Tue Aug  9 16:14:40 2022
```

**Comment by nicogodet** [comment]
I'm very busy right now but I plan to finish this by the end of the week.

**Comment by agiudiceandrea** [comment]
Please see also https://github.com/qgis/QGIS-Website/issues/947#issuecomment-968061113

**Comment by nicogodet** [comment]
@agiudiceandrea I planed to fix this too 😉

File: source/site/forusers/alldownloads.rst [hunk 1]
```diff
@@ -116,14 +116,23 @@ First install some tools you will need for this instructions::
 Now install the QGIS Signing Key, so QGIS software from 
 the QGIS repo will be trusted and installed::
 
- wget -qO - https://qgis.org/downloads/qgis-2022.gpg.key | sudo gpg --no-default-keyring --keyring gnupg-ring:/etc/apt/trusted.gpg.d/qgis-archive.gpg --import
- sudo chmod a+r /etc/apt/trusted.gpg.d/qgis-archive.gpg
+ sudo mkdir -p /etc/apt/keyrings && sudo chmod 755 /etc/apt/keyrings
+ sudo wget -O/etc/apt/keyrings/qgis-archive-keyring.gpg https://download.qgis.org/downloads/qgis-archive-keyring.gpg
```
Comment (agiudiceandrea) [inline source/site/forusers/alldownloads.rst]:
```suggestion
 sudo wget -O /etc/apt/keyrings/qgis-archive-keyring.gpg https://download.qgis.org/downloads/qgis-archive-keyring.gpg
```

File: source/site/forusers/alldownloads.rst [hunk 2]
```diff
@@ -249,54 +259,61 @@ Supported distribution versions:
...
- deb     https://qgis.org/debian unstable main
- deb-src https://qgis.org/debian unstable main
+  sudo mkdir -p /etc/apt/keyrings && sudo chmod 755 /etc/apt/keyrings
+  sudo cp qgis-archive-keyring.gpg /etc/apt/keyrings/qgis-archive-keyring.gpg
 
-After that type the commands below to install QGIS::
+Alternatively you can download the key directly without manual verification::
+        
+  sudo mkdir -p /etc/apt/keyrings && sudo chmod 755 /etc/apt/keyrings
+  sudo wget -O/etc/apt/keyrings/qgis-archive-keyring.gpg https://download.qgis.org/downloads/qgis-archive-keyring.gpg
```
Comment (agiudiceandrea) [inline source/site/forusers/alldownloads.rst]:
```suggestion
  sudo wget -O /etc/apt/keyrings/qgis-archive-keyring.gpg https://download.qgis.org/downloads/qgis-archive-keyring.gpg
```

**Review by eshattow — CHANGES_REQUESTED** [review]
1. I think remove
`sudo mkdir -p /etc/apt/keyrings && sudo chmod 755 /etc/apt/keyrings`
and substitute with
`sudo mkdir -m755 -p /etc/apt/keyrings # not needed since apt version 2.4.0 like Debian 12 and Ubuntu 22 or newer`

This comment about apt version 2.4.0 has its source of information for reference in this review per https://metadata.ftp-master.debian.org/changelogs//main/a/apt/apt_2.5.2_changelog:
```
apt (2.4.0) unstable; urgency=medium
... * Install an empty /etc/apt/keyrings directory.
```
2. It is my opinion that qgis-archive-keyring.gpg file must be verified (in the meaning that it is a valid file and operable not corrupted in download) before installing to the system. Remove that suggestion about skipping verification from the document and it actually makes things clearer to understand also avoiding wget as root.

3. There are some (minor) whitespace changes, is that intentional?

**Review by eshattow — APPROVED** [review]

**Comment by agiudiceandrea** [comment]
Was this PR already merged? https://github.com/qgis/QGIS-Website/commit/1ee53d3a34042e3a3e65cada560d55916db9844b

**Comment by nicogodet** [comment]
I guess
It's live on https://www.qgis.org/fr/site/forusers/alldownloads.html#debian-ubuntu

**Pull request closed without merging**